shelf/utils.py

```diff
@@ -20,6 +21,23 @@ def create_path(*args):
     return full_path
 
 
+def get_host_from_uri(uri):
+    """
+        Gets the host (which is actually the protocol + the
+        hostname) from a full URI.
+
+        Args:
+            uri(string)
+
+        Returns:
+            string
+    """
+    parts = urlparse.urlparse(uri)
+    host = parts.scheme + "://" + parts.hostname
```

Comment:
crap. I think there was a reason but I can't remember what it is now. Let me test around with that.